src/Core/Helpers/ObservableCollectionExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Core.Helpers
+{
+    public static class ObservableCollectionExtensions
+    {
+        /// <summary>
+        /// Removes all items from the collection and raises NotifyCollectionChanged for every removed item
+        /// </summary>
+        public static void RemoveAll<T>(this ObservableCollection<T> collection)
+        {
+            var allItems = collection.ToList();
```

Comment:
I remember that you mentioned something about deferring the updates to introduce a bulk-change notification. I couldn't find it, except writing our own ObservableCollection that implements it. Is that what you meant?

Reply:
No, it`s on the collection view: `CollectionView.DeferRefresh()`